client/signup/steps/domains/index.jsx

```diff
@@ -120,6 +120,11 @@ const DomainsStep = React.createClass( {
 					domain: suggestion.domain_name,
 					productSlug: suggestion.product_slug
 				} )
+				: undefined,
+			privacyItem = isPurchasingItem && this.props.domainsWithPlansOnly
```

Comment:
The problem here is that we don't have access to a cart yet, so additional checks are hard. What we might want to do, regardless of the state is to auto-add privacy. 

Opinions? @bisko @meremagee